src/v/kafka/protocol/schemata/generator.py

```diff
@@ -805,60 +862,82 @@ class response;
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 
-{% macro version_guard(field) %}
-{%- set cond = field.versions().guard() %}
-{%- if cond %}
+{% macro version_guard(field, flex) %}
+{%- set guard_enum = field.versions().guard_enum %}
+{%- set e, cond = field.versions().guard(flex, first_flex) %}
+{%- if e == guard_enum.NO_GUARD %}
+{{- caller() }}
+{%- elif e == guard_enum.NO_SOURCE %}
+{%- elif e == guard_enum.GUARD %}
 if ({{ cond }}) {
 {{- caller() | indent }}
 }
 {%- else %}
-{{- caller() }}
+#error unexpected enumeration case encountered
```

Comment:
i'm not sure i understand this? it sort of implies that the code generator is handling a case that wouldn't be valid?

Reply:
Yes, i.e. theres a bug in the code generator itself

Reply:
since we are generating the code, I'm not sure why this would be the case

Reply:
its akin to adding a `builtin_unreachable()` in the `default:` case of a switch statement where every case is handled